src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -0,0 +1,262 @@
+public abstract class Token {
+  private static final String CALLING_MODULE_HEADER = "X-Okapi-Calling-Module";
+  private boolean usesDummyPermissionsSource;
+  protected static final String UNDEFINED_USER_NAME = "UNDEFINED_USER__";
+  protected String source;
+
+  /**
+   * Gets the claims for this token.
+   * @return A JsonObject containing the claims for this token.
+   */
+  public JsonObject getClaims() {
+    return claims;
+  }
+  protected JsonObject claims;
+
+  /**
+   * All implementors of Token are required to implement this method. Validation should
+   * consist of all operations necessary to determine whether the token can authorize
+   * the request.
+   * @param request The http request context that where the token is being provided.
+   * @return If the token is valid, implementors should return a Future with a Token object.
+   * If the token is not valid, implementors should return a Future with a
+   * TokenValidationException.
+   * @see TokenValidationException
+   */
+  protected abstract Future<Token> validate(HttpServerRequest request);
+
+  /** 
+   * Validates the provided token. Validation includes checking everything needed
+   * to determine whether the token should be authorized, including the signature and
+   * any special validation required by its type.
+   * @param sourceToken The JWT or JWE token to validate.
+   * @param tokenCreator The TokenCreator used to encode the token.
+   * @param request The request in the http context where the token is being provided.
+   * @return Future<Token> A Future containing the Token if it has passed validation.
+   * The Future may also contain a TokenValidationException if the validation has failed.
+   * @see TokenValidationException.
+   */
+  public static Future<Token> validate(String sourceToken,
+                                       TokenCreator tokenCreator,
+                                       HttpServerRequest request) {
+    Token token = null;
+    try {
+      token = parse(sourceToken, tokenCreator);
+    } catch (TokenValidationException e) {
+      return Future.failedFuture(e);
+    } catch (Exception e) {
+      return Future.failedFuture(new TokenValidationException("Unexpected token parse exception", e, 500));
+    }
+
+    // Call the validate implementation of the underlying token type (AccessToken, RefreshToken, etc.).
+    // See those classes for the validation logic specific to each type.
+    return token.validate(request);
+  }
+
+  /**
+   * Some tokens require that the dummy permissions source be used. This will return
+   * true when that is the case.
+   * @return True if the dummy permissions source should be used.
+   */
+  public boolean shouldUseDummyPermissionsSource() {
+    // Dummy tokens are not the only type of tokens that require the use of this
+    // so checking the token type alone isn't enough. We also have to check the sub claim.
+    if ((claims.getBoolean("dummy") != null &&
+         claims.getBoolean("dummy"))
```

Comment:
```suggestion
    if (Boolean.TRUE.equals(claims.getBoolean("dummy"))
```

Reply:
This should be done. Think Adam beat you to it.